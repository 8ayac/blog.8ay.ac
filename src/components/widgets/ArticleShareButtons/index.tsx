import { config } from '@/site.config';
import { Facebook } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Facebook';
import { Hatena } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Hatena';
import { Line } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Line';
import { Pocket } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Pocket';
import { Reddit } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Reddit';
import { Twitter } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Twitter';
import { getArticlePagePath } from '@/src/shared/utils';
import { Article } from '@/src/types';
import styled from '@emotion/styled';
import React from 'react';
import urljoin from 'url-join';

const S = {
  ComponentWrapperSection: styled.section``,

  ButtonListWrapperFlexDiv: styled.div`
    display: flex;
  `,

  ButtonWrapperDiv: styled.div`
    margin-right: 1rem;
    margin-bottom: 0.5rem;
    filter: drop-shadow(
      2px 2px 1px
        ${(props) =>
          props.theme.color.articleShareButtons.buttonWrapperDiv.shadow}
    );

    :hover {
      filter: brightness(120%);
    }
  `,
};

export const ArticleShareButtons: React.FC<{
  article: Article;
  size?: number;
}> = ({ article, size = 32 }) => {
  const shareUrl = urljoin(config.site.rootUrl, getArticlePagePath(article));
  const shareTitle = article.attributes.title;
  const shareButtons = (buttonSize: number) => [
    <Facebook key="facebook" shareUrl={shareUrl} size={buttonSize} />,
    <Twitter
      key="twitter"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Hatena
      key="hatena"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Line
      key="line"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Reddit
      key="reddit"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Pocket
      key="pocket"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
  ];

  return (
    <>
      <S.ComponentWrapperSection>
        <S.ButtonListWrapperFlexDiv>
          {shareButtons(size).map((shareButton) => (
            <S.ButtonWrapperDiv key={shareButton.key}>
              {shareButton}
            </S.ButtonWrapperDiv>
          ))}
        </S.ButtonListWrapperFlexDiv>
      </S.ComponentWrapperSection>
    </>
  );
};
