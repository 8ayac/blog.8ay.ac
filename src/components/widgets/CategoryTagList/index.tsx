import { CategoryTag } from '@/src/components/elements/CategoryTag';
import { FlexContainerDiv } from '@/src/shared/styles/abstractStyledComponents';
import styled from '@emotion/styled';
import { faTags } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled(FlexContainerDiv)`
    flex-flow: row wrap;

    > * {
      margin-right: 0.7rem;
      margin-bottom: 0.7rem;
    }
  `,

  PreIconWrapperDiv: styled(FlexContainerDiv)`
    align-items: center;
    color: ${(props) => props.theme.color.text.primaryLight};
  `,
};

export const CategoryTagList: React.FC<{
  tags: string[];
  withPreIcon?: boolean;
}> = ({ tags, withPreIcon = false }) => (
  <>
    <S.ComponentWrapperDiv>
      {withPreIcon && (
        <S.PreIconWrapperDiv>
          <FontAwesomeIcon icon={faTags} />
        </S.PreIconWrapperDiv>
      )}
      {[...new Set(tags)].sort().map((tag) => (
        <CategoryTag key={tag} name={tag} />
      ))}
    </S.ComponentWrapperDiv>
  </>
);
