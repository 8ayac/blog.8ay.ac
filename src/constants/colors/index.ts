export * from './blue';
export * from './green';
export * from './yellow';
