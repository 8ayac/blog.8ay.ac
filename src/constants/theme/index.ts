import * as colors from '@/src/constants/colors';

export const theme = {
  color: {
    blue: colors.blue,
    green: colors.green,
    yellow: colors.yellow,

    primary: colors.blue.base,
    primaryLight: colors.blue.light,
    primaryDark: colors.blue.dark,

    text: {
      primary: 'hsla(0, 0%, 15%)',
      primaryLight: 'hsla(0, 0%, 40%)',
      secondary: colors.blue.dark2,
      accent: colors.green.dark,
    },
  },
};
