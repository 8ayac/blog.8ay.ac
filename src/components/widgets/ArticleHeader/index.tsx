import { BodyHeader } from '@/src/components/elements/BodyHeader';
import { DateTime } from '@/src/components/elements/DateTime';
import { CategoryTagList } from '@/src/components/widgets/CategoryTagList';
import { theme } from '@/src/constants/theme';
import { FlexContainerDiv } from '@/src/shared/styles/abstractStyledComponents';
import { getArticlePagePath } from '@/src/shared/utils';
import { Article } from '@/src/types';
import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { faTags } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const S = {
  DateTimeWrapperDiv: styled(FlexContainerDiv)`
    flex-flow: row wrap;
    align-items: center;
    margin-bottom: 1rem;

    > * {
      margin-right: 1rem;
    }
  `,

  CategoryListWrapperDiv: styled.div`
    margin-top: 2rem;
    margin-bottom: 0.5rem;
  `,
};

export const ArticleHeader: React.FC<{ article: Article }> = ({ article }) => {
  const date = (
    <S.DateTimeWrapperDiv>
      <DateTime date={article.publishedAt} description="Published At" />
      <DateTime date={article.updatedAt} description="Updated At" />
    </S.DateTimeWrapperDiv>
  );
  const tags = (
    <S.CategoryListWrapperDiv>
      <FontAwesomeIcon
        icon={faTags}
        css={css`
          margin-right: 1rem;
          margin-left: 0.5rem;
          font-size: 1em;
          color: ${theme.color.text.primaryLight};
        `}
      />
      <CategoryTagList tags={article.tags as string[]} />
    </S.CategoryListWrapperDiv>
  );

  return (
    <BodyHeader
      title={article.title}
      linkTo={getArticlePagePath(article)}
      beforeTitle={date}
      afterTitle={tags}
    />
  );
};
