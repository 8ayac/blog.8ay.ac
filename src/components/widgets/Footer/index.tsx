import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import React from 'react';

const S = {
  Footer: styled.footer`
    padding: 3rem 0;
    font-size: 1.5rem;
    text-align: center;
  `,

  CopyRightParagraph: styled.p``,

  AboutGAParagraph: styled.p`
    color: ${theme.color.text.primaryLight};
  `,
};

export const Footer: React.FC = () => (
  <S.Footer>
    <S.CopyRightParagraph>
      Â© 2021 blog.8ay.ac All Rights Reserved.
    </S.CopyRightParagraph>
    <S.AboutGAParagraph>
      This website uses Google Analytics, a web analytics service by Google.
      Google Analytics uses cookies, which are placed on your computer to help
      the website analyze how you use the site. You can prevent the the
      collection of your data by disabling cookies in your browser.
    </S.AboutGAParagraph>
  </S.Footer>
);
