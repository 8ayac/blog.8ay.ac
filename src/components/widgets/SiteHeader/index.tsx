import { BlogTitle } from '@/src/components/elements/BlogTitle';
import { SocialLinkWithIcon } from '@/src/components/elements/SocialLinkWithIcon';
import { BLOG_SUBTITLE, BLOG_TITLE } from '@/src/constants/site';
import { theme } from '@/src/constants/theme';
import { FlexContainer } from '@/src/shared/styles/abstractStyledComponents';
import styled from '@emotion/styled';
import { faTwitter } from '@fortawesome/free-brands-svg-icons';
import React from 'react';

const S = {
  Header: styled.header`
    position: sticky;
    top: 0;
    z-index: 100;
    height: 3.7rem;
    padding: 0.5rem 1rem;
    background-color: ${theme.color.green.dark2};
    border-bottom: 1px solid ${theme.color.green.light};
  `,

  HeaderInner: styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 100%;
    color: white;
  `,

  TitleWrapper: styled(FlexContainer)`
    align-items: center;
  `,

  NavWrapper: styled(FlexContainer)`
    align-items: center;
  `,
};

export const SiteHeader: React.FC = () => (
  <S.Header>
    <S.HeaderInner>
      <S.TitleWrapper>
        <BlogTitle title={BLOG_TITLE} subtitle={BLOG_SUBTITLE} />
      </S.TitleWrapper>

      <S.NavWrapper>
        <SocialLinkWithIcon
          url="http://twitter.com/8ayac"
          description="8ayac"
          icon={faTwitter}
        />
      </S.NavWrapper>
    </S.HeaderInner>
  </S.Header>
);
