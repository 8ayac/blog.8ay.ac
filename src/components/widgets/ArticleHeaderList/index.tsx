import { ArticleHeader } from '@/src/components/widgets/ArticleHeader';
import { Article } from '@/src/types';

import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    display: grid;
    grid-gap: 5%;
    margin: 0 auto;
  `,

  ArticleHeaderWrapperDiv: styled.div`
    h1 {
      font-size: 2.3rem;
    }
  `,
};

export const ArticleHeaderList: React.FC<{ articles: Article[] }> = ({
  articles,
}) => (
  <>
    <S.ComponentWrapperDiv>
      <S.ArticleHeaderWrapperDiv>
        {articles
          .sort(
            (a, b) =>
              new Date(a.publishedAt).getTime() -
              new Date(b.publishedAt).getTime()
          )
          .reverse()
          .map((article) => (
            <ArticleHeader key={article.attributes.id} article={article} />
          ))}
      </S.ArticleHeaderWrapperDiv>
    </S.ComponentWrapperDiv>
  </>
);
