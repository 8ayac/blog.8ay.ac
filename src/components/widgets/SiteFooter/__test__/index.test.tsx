import { SiteFooter } from '@/src/components/widgets/SiteFooter';
import { theme } from '@/src/constants/theme';
import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';
import React from 'react';

describe('SiteFooter', () => {
  const wrapper = mount(
    <ThemeProvider theme={theme}>
      <SiteFooter />
    </ThemeProvider>
  );
  it('is rendered correctly to match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('has proper style rules', () => {
    test('in Footer', () => {
      expect(wrapper.find('Footer')).toHaveStyleRule('padding', '3rem 0');
      expect(wrapper.find('Footer')).toHaveStyleRule('margin-top', '10rem');
      expect(wrapper.find('Footer')).toHaveStyleRule('font-size', '1.5rem');
      expect(wrapper.find('Footer')).toHaveStyleRule('text-align', 'center');
    });

    test.skip('in CopyRightParagraph', () => null);

    test('in AboutGAParagraph', () => {
      expect(wrapper.find('AboutGAParagraph')).toHaveStyleRule(
        'color',
        theme.color.text.primaryLight
      );
    });
  });
});
