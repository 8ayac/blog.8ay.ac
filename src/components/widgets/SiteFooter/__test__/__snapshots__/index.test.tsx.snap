// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteFooter is rendered correctly to match the snapshot 1`] = `
.emotion-0 {
  padding: 3rem 0;
  margin-top: 10rem;
  font-size: 1.5rem;
  text-align: center;
}

.emotion-6 {
  color: hsla(0, 0%, 40%);
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "articleShareButtons": Object {
          "buttonWrapperDiv": Object {
            "shadow": "hsla(212, 72%, 75%)",
          },
        },
        "blogTitle": Object {
          "logoWrapperAnchor": Object {
            "textOnHover": "hsla(60, 72%, 65%)",
          },
        },
        "bodyHeader": Object {
          "titleAnchor": Object {
            "shadow": "hsla(148, 72%, 85%)",
          },
        },
        "categoryTag": Object {
          "anchor": Object {
            "bg": "hsla(212, 72%, 45%)",
            "bgOnHover": "hsla(212, 72%, 55%)",
            "shadow": "hsla(212, 72%, 75%)",
            "text": "white",
          },
        },
        "common": Object {
          "border": Object {
            "primary": "hsla(0, 0%, 50%)",
            "primaryLight": "hsla(0, 0%, 75%)",
          },
          "text": Object {
            "accent": "hsla(148, 72%, 45%)",
            "primary": "hsla(0, 0%, 15%)",
            "primaryLight": "hsla(0, 0%, 40%)",
            "secondary": "hsla(212, 72%, 35%)",
          },
        },
        "dateTime": Object {
          "isoDateTime": Object {
            "text": "hsla(212, 72%, 35%)",
          },
        },
        "global": Object {
          "body": Object {
            "bg": "hsla(60, 72%, 55%)",
          },
          "table": Object {
            "tr": Object {
              "bg2n": "hsla(148, 72%, 95%)",
            },
          },
        },
        "markdownRenderer": Object {
          "convertedMdBody": Object {
            "blockquote": Object {
              "border": "hsla(212, 72%, 95%)",
            },
          },
        },
        "siteHeader": Object {
          "header": Object {
            "bg": "hsla(148, 72%, 25%)",
            "shadow": "hsla(148, 72%, 15%)",
          },
          "innerWrapperFlexContainerDiv": Object {
            "shadow": "hsla(148, 72%, 15%)",
          },
        },
        "socialLinkWithIcon": Object {
          "socialLinkAnchor": Object {
            "textOnHover": "hsla(60, 72%, 55%)",
          },
        },
      },
    }
  }
>
  <SiteFooter>
    <Footer>
      <footer
        className="emotion-0 emotion-1"
      >
        <CopyRightParagraph>
          <p
            className="emotion-2 emotion-3"
          >
            Â© 2021
             
            <AuthorAnchor
              href="https://github.com/8ayac/blog.8ay.ac"
            >
              <a
                className="emotion-4 emotion-5"
                href="https://github.com/8ayac/blog.8ay.ac"
              >
                blog.8ay.ac
              </a>
            </AuthorAnchor>
             
            All Rights Reserved.
          </p>
        </CopyRightParagraph>
        <AboutGAParagraph>
          <p
            className="emotion-6 emotion-7"
          >
            This website uses Google Analytics, a web analytics service by Google. Google Analytics uses cookies, which are placed on your computer to help the website analyze how you use the site. You can prevent the the collection of your data by disabling cookies in your browser.
          </p>
        </AboutGAParagraph>
      </footer>
    </Footer>
  </SiteFooter>
</ThemeProvider>
`;
